### PR TITLE
Added limited role

### LINKED ISSUE #19891: `limited` role/status for new users onto the platform
## Is this feature related to a problem? Please describe.

- As a Forem admin, I want a clear means of establishing who a new user is, so I can manually or automatically review their profile before making their presence more widely known on the platform.

## Describe the solution you’d like

- We currently advertise new users on the platform through a number of means:
  - Round-robin notifications for new user activity for trusted users
  - "Follow" notifications, as users are encouraged to follow other users as part of onboarding
- This approach is problematic for spam accounts, as established users are made aware of a disproportionate amount of spam accounts that otherwise have their impact limited on the platform.
- We will introduce a new `limited` user status, that we can drive functionality off in future issues to limit the impact of these unverified accounts.

## Definition of Done

- [ ] Add a new `limited` user status that is displayed in the UI as "Limited"
  - The status should be appliable by users with the `admin` role
  - The status should be supported by the bulk status assignment tool 
  - When users have the `limited` status applied, the pill should appear on the Member manager in `/admin` in yellow. e.g.image
  - Generally, the status should be displayed in line with the existing statuses elsewhere on the platform

## Additional Context

- Although the new user context is most relevant in this situation, `limited` is a more generic name as it might be used in moderation practices in different contexts.
- We will drive functionality to stop round-robin notifications for `limited` users in https://github.com/forem/forem/issues/19892
- In the future, we may drive other functionality on this setting. For example, the posting frequency limits placed on new user accounts.

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

Added limited role to Role/Status

## Related Tickets & Documents

- Closes #19891

### Output

1. Sign in as admin
2. Visit `/admin/member_manager/users` and select any user to `Add Role`, in that add `Limited` role.
3. Check that this role is visible on the `/admin/member_manager/users` as well as on user-profile too, i.e. `/admin/member_manager/users/<user-id>`
4. Add `limited` role via Bulk assign from `/admin/member_manager/bulk_assign_role`

Screenshot 2023-08-22 at 1 01 22 AM
Screenshot 2023-08-22 at 1 00 12 AM
Screenshot 2023-08-22 at 12 59 59 AM

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
No A11Y testing required here.

## Added/updated tests?

- [x] Yes

### CODE REVIEW

**Comment by Ridhwana** [comment]
@rt4914 Just a headsup that I'll look through this PR later today.

**Review by Ridhwana — CHANGES_REQUESTED** [review]
@rt4914 this looks mostly good, however I think we can rollback that migration.

File: db/schema.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema[7.0].define(version: 2023_08_03_133228) do
+ActiveRecord::Schema[7.0].define(version: 2023_08_21_183447) do
```
[the remark this answers is not in the document]
Comment (Ridhwana) [reply db/schema.rb:13]: @rt4914 are we able to revert the timestamp as well?

File: db/migrate/20230821183447_add_limited_role_to_roles.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class AddLimitedRoleToRoles < ActiveRecord::Migration[7.0]
+  def up
+    Role.create!(name: "limited") unless Role.exists?(name: "limited")
+  end
+
+  def down
+    role = Role.find_by(name: "limited")
+    role&.destroy
+  end
+end
```
[the remark this answers is not in the document]
Comment (rt4914) [reply db/migrate/20230821183447_add_limited_role_to_roles.rb]: Done. Reverted this change

File: db/schema.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (rt4914) [reply db/schema.rb]: Somehow this got reverted locally. Updated now.

**Review by Ridhwana — APPROVED** [review]
Thanks for those changes @rt4914, this LGTM 🚀

**Review by maestromac — APPROVED** [review]
LGTM!

**Pull request merged**